sdk/identity/azure-identity/tests/test_auth_code_async.py

```diff
@@ -32,6 +32,32 @@
     assert policy.on_request.called
 
 
+@pytest.mark.asyncio
+async def test_close():
+    transport = AsyncMockTransport()
+    credential = AuthorizationCodeCredential(
+        "tenant-id", "client-id", "auth-code", "http://localhost", transport=transport
+    )
+
+    await credential.close()
+
+    assert transport.__aexit__.call_count == 1
+
+
+@pytest.mark.asyncio
+async def test_context_manager():
+    transport = AsyncMockTransport()
+    credential = AuthorizationCodeCredential(
+        "tenant-id", "client-id", "auth-code", "http://localhost", transport=transport
+    )
+
+    async with credential:
+        assert transport.__aenter__.call_count == 1
+
+    assert transport.__aenter__.call_count == 1
+    assert transport.__aexit__.call_count == 1
+
```

Comment:
Can you add credential.close() after the context manager to make sure double close does not cause problems?

Reply:
I could, but the credential just wraps the transport's methods, which these tests already verify. It seems like what you really want is to ensure double closing the transport is okay. I want that too, but to me it's a test for the real transport implementations.